aws/resource_aws_directory_service_log_subscription.go

```diff
@@ -0,0 +1,98 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directoryservice"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDirectoryServiceLogSubscription() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDirectoryServiceLogSubscriptionCreate,
+		Read:   resourceAwsDirectoryServiceLogSubscriptionRead,
+		Update: nil,
+		Delete: resourceAwsDirectoryServiceLogSubscriptionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"directory_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"log_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDirectoryServiceLogSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+	dsconn := meta.(*AWSClient).dsconn
+
+	directoryId := d.Get("directory_id")
+	logGroupName := d.Get("log_group_name")
+
+	input := directoryservice.CreateLogSubscriptionInput{
+		DirectoryId:  aws.String(directoryId.(string)),
+		LogGroupName: aws.String(logGroupName.(string)),
+	}
+
+	_, err := dsconn.CreateLogSubscription(&input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(directoryId.(string))
+
+	return resourceAwsDirectoryServiceLogSubscriptionRead(d, meta)
+}
+
+func resourceAwsDirectoryServiceLogSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	dsconn := meta.(*AWSClient).dsconn
+
+	directoryId := d.Id()
+
+	input := directoryservice.ListLogSubscriptionsInput{
+		DirectoryId: aws.String(directoryId),
+	}
+
+	out, err := dsconn.ListLogSubscriptions(&input)
+	if err != nil {
+		return err
+	}
+
+	if len(out.LogSubscriptions) == 0 {
+		log.Printf("[WARN] No log subscriptions for directory %s found", directoryId)
+		return nil
```

Comment:
This should call `d.SetId("")` before returning, which will trigger Terraform to automatically recreate the resource

```suggestion
		d.SetId("")
		return nil
```